pytest_fixtures/satellite_auth.py

```diff
@@ -149,6 +152,31 @@ def enroll_idm_and_configure_external_auth():
     )
 
 
+@fixture()
+def rhsso_setting_setup(request):
+    """Update the RHSSO setting and revert it in cleanup"""
+    update_rhsso_settings_in_satellite()
+
+    def rhsso_setting_cleanup():
+        update_rhsso_settings_in_satellite(revert=True)
+
+    request.addfinalizer(rhsso_setting_cleanup)
```

Comment:
If you're going to define a function for cleanup, you can decorate that function instead of having to then explicitly call it here.

```
@request.addfinalizer
def _cleanup():
    update_rhsso_settings_in_satellite(revert=True)
```

That being said, for such a simple call, it could be done via a lambda passed to `request.addfinalizer`.

And _that_ being said, here I would just yield. Its my preferred method for doing fixture cleanup when I'm not concerned about runtime exceptions in a complex fixture.

```
@pytest.fixture
def rhsso_settings():
    update_rhsso_settings_in_satellite()
    yield
     update_rhsso_settings_in_satellite(revert=True)
```